packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts

```diff
@@ -0,0 +1,52 @@
+import { objectValues } from '@datadog/browser-core'
+
+type SelectorStrategy = (
+  element: Element | HTMLElement | HTMLInputElement | HTMLSelectElement
+) => string | undefined | null
+
+export function getSelectorFromElement(targetElement: Element): string {
+  const strategies: { [name: string]: SelectorStrategy } = {
+    getIDSelector: (element) => {
+      const isUnique = element.id && element.ownerDocument.body.querySelectorAll(`#${element.id}`).length === 1
+
+      if (isUnique) return `#${element.id}`
+    },
+    getClassSelector: (element) => {
+      const orderedClassList = Array.from(element.classList).sort()
+      const siblings = Array.from(element.parentElement!.children).filter((child) => child !== element)
+      const classUniqueAmongSiblings = siblings.every(
+        (sibling) =>
+          sibling.tagName !== element.tagName ||
+          orderedClassList.some((className) => !sibling.classList.contains(className))
+      )
+
+      if (classUniqueAmongSiblings)
+        return `${element.tagName}${orderedClassList.map((className) => `.${className}`).join('')}`
+    },
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I had a quick look but it probably worth iterating on this part:

- tests seems a bit light
- not sure why sorting is needed (perf concern)
- what if `sibling.classList = [...orderedClassList, otherClass]`

Reply:
Sorting is needed to always have the same css path string regardless of the class name order.